zha/application/gateway.py

```diff
@@ -204,6 +206,14 @@ def get_application_controller_data(self) -> tuple[ControllerApplication, dict]:
             CONF_DEVICE_FLOW_CONTROL: self.config.config.coordinator_configuration.flow_control,
         }
 
+        if (
+            self.config.country_code is not None
+            and CONF_NWK_COUNTRY_CODE not in app_config.get(CONF_NWK, {})
+        ):
+            app_config.setdefault(CONF_NWK, {})[CONF_NWK_COUNTRY_CODE] = (
+                self.config.country_code
+            )
```

Comment:
Totally fine with me, but this just wasn't a pattern I see a lot with the `setdefault` and directly setting the key then.

Alternative could be to make it more explicit, but both are fine:
```suggestion
            app_config.setdefault(CONF_NWK, {})
            app_config[CONF_NWK][CONF_NWK_COUNTRY_CODE] = self.config.country_code
```